pkg/clients/dynatrace/settings.go

```diff
@@ -0,0 +1,174 @@
+package dynatrace
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/utils"
+	"github.com/pkg/errors"
+)
+
+type monitoredEntitiesResponse struct {
+	Entities   []MonitoredEntity `json:"entities"`
+	TotalCount int               `json:"totalCount"`
+	PageSize   int               `json:"pageSize"`
+}
+
+type MonitoredEntity struct {
+	EntityId    string `json:"entityId"`
+	DisplayName string `json:"displayName"`
+	LastSeenTms int64  `json:"lastSeenTms"`
+}
+
+type GetSettingsResponse struct {
+	TotalCount int `json:"totalCount"`
+}
+
+type postSettingsResponse struct {
+	ObjectId string `json:"objectId"`
+}
+
+type getSettingsErrorResponse struct {
+	ErrorMessage getSettingsError `json:"error"`
+}
+
+type getSettingsError struct {
+	Message              string
+	ConstraintViolations constraintViolations `json:"constraintViolations,omitempty"`
+	Code                 int
+}
+
+type constraintViolations []struct {
+	ParameterLocation string
+	Location          string
+	Message           string
+	Path              string
+}
+
+func (dtc *dynatraceClient) GetMonitoredEntitiesForKubeSystemUUID(ctx context.Context, kubeSystemUUID string) ([]MonitoredEntity, error) {
+	if kubeSystemUUID == "" {
+		return nil, errors.New("no kube-system namespace UUID given")
+	}
+
+	req, err := createBaseRequest(ctx, dtc.getEntitiesUrl(), http.MethodGet, dtc.apiToken, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	q := req.URL.Query()
+	q.Add("pageSize", "500")
+	q.Add("entitySelector", fmt.Sprintf("type(KUBERNETES_CLUSTER),kubernetesClusterId(%s)", kubeSystemUUID))
+	q.Add("from", "-365d")
+	q.Add("fields", "+lastSeenTms")
+	req.URL.RawQuery = q.Encode()
+
+	res, err := dtc.httpClient.Do(req)
+	if err != nil {
+		log.Info("check if ME exists failed")
+
+		return nil, err
+	}
+
+	defer utils.CloseBodyAfterRequest(res)
```

Comment:
if i'm not mistaken we should close body, and then check error, because in this case if error occurs, body is still open. Check edge connect client code.

Reply:
fixed in 5df7a83 (old code that was just moved)